storage/innobase/fil/fil0fil.cc

```diff
@@ -7353,9 +7369,9 @@ fil_set_encryption(
 	byte*			key,
 	byte*			iv)
 {
-	ut_ad(!is_system_or_undo_tablespace(space_id));
+	ut_ad(!is_shared_system_tablespace(space_id));
```

Comment:
Both assertion and subsequent `if(...) return` with the same condition

Reply:
Yep. And it was so in upstream actually. It means in debug build server will crash and in release build it will refuse to create table.